homeassistant/helpers/event.py

```diff
@@ -599,40 +620,64 @@ def async_remove(self) -> None:
         self._cancel_entities_listener()
 
     @callback
-    def async_refresh(self, variables: Any = _UNCHANGED) -> None:
+    def async_refresh(self) -> None:
         """Force recalculate the template."""
-        if variables is not _UNCHANGED:
-            self._variables = variables
         self._refresh(None)
 
     @callback
     def _refresh(self, event: Optional[Event]) -> None:
-        self._info = self._template.async_render_to_info(self._variables)
-        self._update_listeners()
-        self._last_info = self._info
+        entity_id = event and event.data.get(ATTR_ENTITY_ID)
+        updates = []
+        info_changed = False
 
-        try:
-            result: Union[str, TemplateError] = self._info.result
-        except TemplateError as ex:
-            result = ex
+        for template, variables in self._templates:
+            if (
+                entity_id
+                and len(self._last_info) > 1
+                and not self._last_info[template].filter_lifecycle(entity_id)
+            ):
+                continue
 
-        # Check to see if the result has changed
-        if result == self._last_result:
-            return
+            self._info[template] = template.async_render_to_info(variables)
+            info_changed = True
 
-        if isinstance(result, TemplateError) and isinstance(
-            self._last_result, TemplateError
-        ):
+            try:
+                result: Union[str, TemplateError] = self._info[template].result
+            except TemplateError as ex:
+                result = ex
+
+            last_result = self._last_result.get(template)
+
+            # Check to see if the result has changed
+            if result == last_result:
+                continue
+
+            if isinstance(result, TemplateError) and isinstance(
+                last_result, TemplateError
+            ):
+                continue
+
+            updates.append((template, last_result, result))
```

Comment:
Do you think that we'll ever want to add something here ? Because if we want to do that, we should wrap it in a namedtuple/data class

Reply:
I think we would probably change the inputs if we were going to add to the outputs.  Maybe make the input and outputs data classes?

Reply:
Sounds good.

Reply:
done